examples/ArduinoWiFi101/ArduinoWiFi101.ino

```diff
@@ -62,10 +66,6 @@ void loop() {
   }
 }
 
-void messageReceived(String topic, String payload, char * bytes, unsigned int length) {
-  Serial.print("incoming: ");
-  Serial.print(topic);
-  Serial.print(" - ");
-  Serial.print(payload);
-  Serial.println();
+void messageReceived(String topic, String payload) {
```

Comment:
I assume the 0x00 terminator would make String inconvenient to handling binary data. I'm beginning to see the sense of a general Buffer class for Arduino like you mentioned, though, @sandeepmistry. 

@256dpi, on Sandeep's suggestion I just used this lib for an example in a new edition of a book I'm doing, and I like it a lot. Example [here](https://github.com/tigoe/MakingThingsTalk2/tree/master/3rd_edition/chapter10/catCamDeux/AirConditionerArduino). I'm happy to comment on things if it's useful to you. Overall I agree with his comments, though, except may be the const String& bit, because I can see use cases where you might assemble or disassemble the String during the sketch.